packages/hash/api/src/graphql/typeDefs/ontology/entity-type.typedef.ts

```diff
@@ -0,0 +1,65 @@
+import { gql } from "apollo-server-express";
+
+export const entityTypeTypedef = gql`
+  scalar EntityType
+
+  type PersistedEntityType {
+    """
+    The specific versioned URI of the entity type
+    """
+    entityTypeVersionedUri: String!
+    """
+    The user who created the entity type
+    """
+    accountId: ID!
+    """
+    The entity type
+    """
+    entityType: EntityType!
+    # TODO: we might need something like
+    # "referencedPropertyTypes: [PersistedPropertyType!]"
+  }
+
+  extend type Query {
+    """
+    Get all entity types at their latest version.
+    """
+    getAllLatestEntityTypes: [PersistedEntityType!]!
+
+    """
+    Get a entity type by its versioned URI.
+    """
+    getEntityType(entityTypeVersionedUri: String!): PersistedEntityType!
+  }
+
+  extend type Mutation {
+    """
+    Create a entity type.
+    """
+    createEntityType(
+      """
+      accountId refers to the account to create the entity type in.
+      """
+      accountId: ID
+      entityType: EntityType!
+    ): PersistedEntityType!
+
+    """
+    Update a entity type.
+    """
+    updateEntityType(
+      """
+      accountId refers to the account to update the entity type in.
+      """
+      accountId: ID!
```

Comment:
For consistency this should be optional
```suggestion
    updateEntityType(
      """
      accountId refers to the account to update the entity type in.
      """
      accountId: ID
```

Reply:
Ah my bad, missed the optionals